src/irmin-pack/inode_intf.ml

```diff
@@ -102,14 +109,18 @@ module type Internal = sig
       (** {1 Concrete trees} *)
 
       (** The type for pointer kinds. *)
-      type kind = Contents | Contents_x of metadata | Node [@@deriving irmin]
+      type kinded_key =
+        | Contents of contents_key
+        | Contents_x of metadata * contents_key
+        | Node of node_key
+      [@@deriving irmin]
 
-      type entry = { name : step; kind : kind; hash : hash } [@@deriving irmin]
-      (** The type for entries. *)
+      type entry = { name : step; key : kinded_key } [@@deriving irmin]
+      (** The type of entries. *)
 
       type 'a pointer = { index : int; pointer : hash; tree : 'a }
       [@@deriving irmin]
-      (** The type for pointers. *)
+      (** The type for internal pointers between concrete {!tree}s. *)
```

Comment:
I don't follow - what is the index field? what is the pointer field?

Reply:
This type is part of the `Inode.Concrete` module that is used to serialise inodes for testing purposes.

A `pointer` `p` is a pointer from a parent inode to a child inode. `p.pointer` is the hash of the child. `p.tree` is the child, `p.index` is the index of `p` in the array of points of the parent.